tmt/cli.py

```diff
@@ -1101,6 +1103,8 @@ def runs(context, path, last, id_, keep, **kwargs):
         raise tmt.utils.GeneralError("--keep must not be a negative number.")
     if not os.path.exists(path):
         raise tmt.utils.GeneralError(f"Path '{path}' doesn't exist.")
+    # Set the option to skip to initialize work tree
+    context.params[tmt.utils.PLAN_SKIP_WORKTREE_INIT] = True
```

Comment:
One question (more of a nitpick) -- shouldn't this rather go into the logic of the `Clean` class? If we imagine that the `Clean` class could be imported somewhere else and then its methods would be called, it would result in rsync being called which is not desired in any case IMO.

Reply:
Good point. And `Clean()` does accept a `context` during initialization so it could handle the context adjustment at one place only?

Reply:
Yes, `Clean()` does accept a `context` during initialization, but `class Clean()` doesn't have its own `__init__` method. If we create an instance of `class Clean()`, it just invokes the [\_\_init\_\_()](https://github.com/teemtee/tmt/blob/main/tmt/utils.py#L267) in  `class Common()`, so I didn't update `class Clean()`.  @FrNecas and @psss, do you think it is worth creating a `__init__()` for `class Clean()`? If yes, the proposed fix will be looked like:

```patch
diff --git a/tmt/base.py b/tmt/base.py
index be17b9e..47c2f3d 100644
--- a/tmt/base.py
+++ b/tmt/base.py
@@ -1897,6 +1897,16 @@ class Status(tmt.utils.Common):
 class Clean(tmt.utils.Common):
     """ A class for cleaning up workdirs, guests or images """
 
+    def __init__(self, parent=None, name=None, workdir=None, context=None):
+        """
+        Initialize name and relation with the parent object
+
+        Always skip to initialize the work tree.
+        """
+        # Set the option to skip to initialize work tree
+        context.params[tmt.utils.PLAN_SKIP_WORKTREE_INIT] = True
+        super().__init__(parent, name, workdir, context)
+
     def images(self):
         """ Clean images of provision plugins """
         self.info('images', color='blue')
diff --git a/tmt/cli.py b/tmt/cli.py
index 3c50517..408b0a5 100644
--- a/tmt/cli.py
+++ b/tmt/cli.py
@@ -1049,8 +1049,6 @@ def clean(context, **kwargs):
     /var/tmp/tmt, if runs are stored elsewhere, the path to them can
     be set using a subcommand (either runs or guests subcommand).
     """
-    # Set the option to skip to initialize work tree
-    context.params[tmt.utils.PLAN_SKIP_WORKTREE_INIT] = True
     clean_obj = tmt.Clean(parent=context.obj, context=context)
     context.obj.clean = clean_obj
     exit_code = 0
@@ -1103,8 +1101,6 @@ def runs(context, path, last, id_, keep, **kwargs):
         raise tmt.utils.GeneralError("--keep must not be a negative number.")
     if not os.path.exists(path):
         raise tmt.utils.GeneralError(f"Path '{path}' doesn't exist.")
-    # Set the option to skip to initialize work tree
-    context.params[tmt.utils.PLAN_SKIP_WORKTREE_INIT] = True
     exit_code = 0
     if not tmt.Clean(parent=context.obj.clean, context=context).runs():
         exit_code = 1
@@ -1137,8 +1133,6 @@ def guests(context, path, last, id_, **kwargs):
             "Options --last and --id cannot be used together.")
     if not os.path.exists(path):
         raise tmt.utils.GeneralError(f"Path '{path}' doesn't exist.")
-    # Set the option to skip to initialize work tree
-    context.params[tmt.utils.PLAN_SKIP_WORKTREE_INIT] = True
     exit_code = 0
     if not tmt.Clean(parent=context.obj.clean, context=context).guests():
         exit_code = 1
@@ -1160,8 +1154,6 @@ def images(context, **kwargs):
     # FIXME: If there are more provision methods supporting this,
     #        we should add options to specify which provision should be
     #        cleaned, similarly to guests.
-    # Set the option to skip to initialize work tree
-    context.params[tmt.utils.PLAN_SKIP_WORKTREE_INIT] = True
     tmt.Clean(parent=context.obj.clean, context=context).images()
```

If it's okay, I'll push the proposed fix above.

Reply:
Yes, this looks better.

Reply:
Thanks @psss!